pkg/tenant/tenants.go

```diff
@@ -138,6 +138,71 @@ func (s *Server) TenantDelete(c *gin.Context) {
 	c.Status(http.StatusOK)
 }
 
+// TenantUpdate will update a tenants records and
+// returns a 200 OK response.
+//
+// Route: /tenant/:tenantID
 func (s *Server) TenantUpdate(c *gin.Context) {
-	c.JSON(http.StatusNotImplemented, "not implemented yet")
+	var (
+		err    error
+		tenant *api.Tenant
+	)
+
+	// TODO: authentication and authorization middleware
+
+	// Get the tenant ID from the URL and return a 400 if the tenant
+	// ID is not a ULID.
+	var tenantID ulid.ULID
+	if tenantID, err = ulid.Parse(c.Param("tenantID")); err != nil {
+		log.Debug().Err(err).Msg("could not parse tenant ulid")
+		c.JSON(http.StatusBadRequest, api.ErrorResponse("could not parse tenant id"))
+		return
+	}
+
+	// Get the specified tenant from the database and return a 500 response
+	// if it can not be retrieved.
+	var t *db.Tenant
+	if t, err = db.RetrieveTenant(c.Request.Context(), tenantID); err != nil {
+		log.Error().Err(err).Msg("could not retrieve tenant")
+		c.JSON(http.StatusNotFound, api.ErrorResponse("could not retrieve tenant"))
+		return
+	}
+
+	// Bind the user request with JSON and return a 400 response if
+	// binding is not successful.
+	if err = c.BindJSON(&tenant); err != nil {
+		log.Warn().Err(err).Msg("could not parse tenant update request")
+		c.JSON(http.StatusBadRequest, api.ErrorResponse("could not bind request"))
+		return
+	}
+
+	// Verify the tenant name exists and return a 400 response if it does not exist.
+	if tenant.Name == "" {
+		c.JSON(http.StatusBadRequest, api.ErrorResponse("tenant name is required"))
+		return
+	}
+
+	// Verify the tenant environment type exists and return a 400 response if it does
+	// not exist.
+	if tenant.EnvironmentType == "" {
+		c.JSON(http.StatusBadRequest, api.ErrorResponse("tenant environment type is required"))
+		return
+	}
+
+	// Prepare update request for insertion into the database.
+	req := &db.Tenant{
+		ID:              t.ID,
+		Name:            tenant.Name,
+		EnvironmentType: tenant.EnvironmentType,
+	}
+
+	// Update tenant in the database and return a 404 response if the
+	// tenant record cannot be updated.
+	if err := db.UpdateTenant(c.Request.Context(), req); err != nil {
+		log.Error().Err(err).Msg("could not save tenant")
+		c.JSON(http.StatusNotFound, api.ErrorResponse("could not update tenant"))
```

Comment:
```suggestion
		c.JSON(http.StatusInternalServerError, api.ErrorResponse("could not update tenant"))
```

Reply:
At this point we have already verified that the tenant exists so if we get an error here then it's probably a server error.